src/App.js

```diff
@@ -1,26 +1,32 @@
-import React, { Component } from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
 import data from './api/movies.json';
 
-export class App extends Component {
-  state = {
-    movies: data,
-  };
+export const App = () => {
+  const [movies, setMovies] = useState(data);
+
+  const addMovie = (newMovie) => {
+    const hasSameMovie = movies.find(movie => newMovie.imdbId === movie.imdbId);
 
-  render() {
-    const { movies } = this.state;
+    if (!hasSameMovie) {
+      const cloneMovies = movies.map(movie => ({ ...movie }));
 
-    return (
-      <div className="page">
-        <div className="page-content">
-          <MoviesList movies={movies} />
-        </div>
-        <div className="sidebar">
-          <FindMovie />
-        </div>
+      setMovies([
+        ...cloneMovies,
+        newMovie]);
```

Comment:
no need to make clone movies. This will be enough
setMovies(movies => [
  ...movies,
  newMovie
]);